src/main/java/com/devqoo/backend/auth/jwt/JwtProvider.java

```diff
@@ -0,0 +1,134 @@
+package com.devqoo.backend.auth.jwt;
+
+import com.devqoo.backend.auth.security.CustomUserDetails;
+import com.devqoo.backend.common.exception.BusinessException;
+import com.devqoo.backend.common.exception.ErrorCode;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.ExpiredJwtException;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.MalformedJwtException;
+import io.jsonwebtoken.UnsupportedJwtException;
+import io.jsonwebtoken.security.Keys;
+import io.jsonwebtoken.security.SignatureException;
+import java.nio.charset.StandardCharsets;
+import java.util.Date;
+import java.util.List;
+import javax.crypto.SecretKey;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.stereotype.Component;
+
+@Component
+public class JwtProvider {
+
+    private final SecretKey accessKey;
+    private final SecretKey refreshKey;
+
+    @Value("${JWT_ACCESS_TIME}")
+    private int ACCESS_EXPIRE_TIME;
+    @Value("${JWT_REFRESH_TIME}")
+    private int REFRESH_EXPIRE_TIME;
+
+    public JwtProvider(
+        @Value("${JWT_ACCESS_SECRET}") String accessKey,
+        @Value("${JWT_REFRESH_SECRET}") String refreshKey
+    ) {
+        this.accessKey = Keys.hmacShaKeyFor(accessKey.getBytes(StandardCharsets.UTF_8));
+        this.refreshKey = Keys.hmacShaKeyFor(refreshKey.getBytes(StandardCharsets.UTF_8));
+    }
+
+    // Access Token 발급
+    public String generateAccessToken(Long userId, String email, String role) {
+        return createToken(userId, email, role, ACCESS_EXPIRE_TIME, accessKey);
+    }
+
+    // Refresh Token 발급
+    public String generateRefreshToken(Long userId, String email, String role) {
+        return createToken(userId, email, role, REFRESH_EXPIRE_TIME, refreshKey);
+    }
+
+    // Token 생성
+    public String createToken(
+        Long userId, String email, String role, int expireTime, SecretKey secretKey
+    ) {
+
+        Date now = new Date();
+
+        return Jwts.builder()
+            .subject(userId.toString())
+            .claim("email", email)
+            .claim("role", role)
+            .issuedAt(now)
+            .expiration(new Date(now.getTime() + expireTime))
+            .signWith(secretKey)
+            .compact();
```

Comment:
```suggestion
    Instant now = Instant.now();
    Instant expiration = now.plusSeconds(expireTime);

    return Jwts.builder()
      .issuedAt(Date.from(now))
      .expiration(Date.from(expiration))
      .signWith(secretKey)
      .compact();

```
조금 더 명확한 것 같아서 이건 어떤가요? + 연산자보다 plus 메소드를 이용해서 조금 더 명확한것같아서

만료 시간을 어떻게 설정하기로 하였는지 공유해주세요.

Reply:
맞아요. 수정 후 공유하도록 하겠습니다.